### PR TITLE
Create database layer description documents

### PR DESCRIPTION
**Pull request by gewenyu99**
## What does this PR do?

Database Layer additions require added documentation and descriptions. This PR aims to add the missing documentation

## Test Plan

(Write your test plan here. If you changed any code, please provide us with clear instructions on how you verified your changes work.)

## Related PRs and Issues

(If this PR is related to any other PR or resolves any issue or related to any issue link all related PR and issues here.)

### Have you read the [Contributing Guidelines on issues](https://github.com/appwrite/appwrite/blob/master/CONTRIBUTING.md)?

(Write your answer here.)

### CODE REVIEW

File: docs/references/database/delete.md [hunk 1]
```diff
@@ -0,0 +1 @@
+Delete a database by its unique ID. Only API keys with with databases.write scope can delete a database.
```
[the remark this answers is not in the document]
Comment (gewenyu99) [reply docs/references/database/delete.md:1]: Addressed

File: docs/references/database/list-documents.md [hunk 2]
```diff
@@ -1 +1 @@
-Get a list of all the user documents. You can use the query params to filter your results. On admin mode, this endpoint will return a list of all of the project's documents. [Learn more about different API modes](/docs/admin).
\ No newline at end of file
+Get a list of all the user's documents in a given collection. You can use the query params to filter your results. On admin mode, this endpoint will return a list of all of documents belonging to the provided collectionId. [Learn more about different API modes](/docs/admin).
```
Comment (gewenyu99) [inline docs/references/database/list-documents.md:1]: > @[lohanidamodar](https://github.com/lohanidamodar) lohanidamodar [16 hours ago](https://github.com/appwrite/appwrite/pull/3391#discussion_r903168202)
as i think now this is misleading. even on admin mode, the endpoint returns list of all of the documents in the collection and database provided, not the project right?


> @[eldadfux](https://github.com/eldadfux) eldadfux [12 hours ago](https://github.com/appwrite/appwrite/pull/3391#discussion_r903275462)
Maybe: Get a list of all the user's documents in the given database.

File: docs/references/database/list.md [hunk 3]
```diff
@@ -0,0 +1 @@
+Get a list of all the user's databases. You can use the query params to filter your results. On admin mode, this endpoint will return a list of all of the project's databases. [Learn more about different API modes](/docs/admin).
```
Comment (gewenyu99) [inline docs/references/database/list.md:1]: >@[lohanidamodar](https://github.com/lohanidamodar) lohanidamodar [17 hours ago](https://github.com/appwrite/appwrite/pull/3391#discussion_r903170438)
We don't have query parameter is only for list documents, we don't yet have it here

>@[eldadfux](https://github.com/eldadfux) eldadfux [12 hours ago](https://github.com/appwrite/appwrite/pull/3391#discussion_r903275828)
Good catch. We should fix it here as well: https://appwrite.io/docs/server/functions?sdk=nodejs-default#functionsList

Functions will be in a separate PR

File: docs/services/databases.md [hunk 4]
```diff
@@ -0,0 +1,7 @@
+The Databases service allows you to create structured collections of documents, query and filter lists of documents, and manage an advanced set of read and write access permissions.
```
Comment (gewenyu99) [inline docs/services/databases.md:1]: >let's rename this file to databases.md also the folder to databases as we have updated database service to databases service

Done

**Review by lohanidamodar — APPROVED** [review]
LGTM
